test/functional/lua/version_spec.lua

```diff
@@ -0,0 +1,570 @@
+        }
+        for _, tc in ipairs(testcases) do
+          it(string.format('for %s: version = %s', tc.desc, tostring(tc.version)), function()
+            eq(version.parse(tc.version, { strict = true }), nil)
+          end)
+        end
+      end)
+
+      describe('raises error', function()
+        local testcases = {
+          { desc = 'no parameters' },
+          { desc = 'nil', version = nil },
+          { desc = 'a number', version = 0 },
+          { desc = 'a float', version = 0.01 },
+          { desc = 'a table', version = {} },
+        }
+        for _, tc in ipairs(testcases) do
+          it(string.format('for %s: version = %s', tc.desc, tostring(tc.version)), function()
+            matches(
+              'is not a valid version',
+              pcall_err(function()
+                version.parse(tc.version, { strict = true })
+              end)
+            )
+          end)
+        end
+      end)
+    end)
+  end)
+
+  describe('eq', function()
+    describe('valid versions', function()
+      local testcases = {
+        {
+          version_1 = '1.0.0',
+          version_2 = '1.0.0',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0',
+          version_2 = 'v1.0.0',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0',
+          version_2 = '1.0',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0',
+          version_2 = '1',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0-alpha',
+          version_2 = '1.0.0-alpha',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0-alpha',
+          version_2 = 'v1.0.0-alpha',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0-alpha',
+          version_2 = '1.0.0-alpha+build.5',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0-alpha.1',
+          version_2 = '1.0.0-alpha.1+build.5',
+          want = true,
+        },
+        {
+          version_1 = '1.0.0-alpha',
+          version_2 = '1.0.0-alpha.1',
+          want = false,
+        },
+        {
+          version_1 = '1.0.0',
+          version_2 = '2.0.0',
+          want = false,
+        },
+      }
+      for _, tc in ipairs(testcases) do
+        it(string.format('returns %s for %s = %s', tc.want, tc.version_1, tc.version_2), function()
+          eq(version.eq(tc.version_1, tc.version_2), tc.want)
+        end)
+      end
+    end)
+
+    describe('errors', function()
+      local testcases = {
+        {
+          version_1 = '',
+          version_2 = '1.0.0',
+        },
+        {
+          version_1 = '1.0.0',
+          version_2 = '',
+        },
+        {
+          version_1 = '',
+          version_2 = '',
+        },
+      }
+      for _, tc in ipairs(testcases) do
+        it(string.format('for %s = %s', tc.version_1, tc.version_2), function()
+          matches(
+            'is not a valid version',
```

Comment:
```suggestion
            'invalid version: foo',
```

Reply:
I think you should comment on line 198 in version.lua